src/riak_kv_pb_coverage.erl

```diff
@@ -0,0 +1,179 @@
+%% following request messages:</p>
+%%
+%% <pre>
+%%  70 - RpbCoverageReq
+%% </pre>
+%%
+%% <p>This service produces the following responses:</p>
+%%
+%% <pre>
+%%  71 - RpbCoverageResp
+%% </pre>
+%% @end
+
+-module(riak_kv_pb_coverage).
+
+-include_lib("riak_pb/include/riak_kv_pb.hrl").
+
+-behaviour(riak_api_pb_service).
+-compile(export_all).
+-export([init/0,
+         decode/2,
+         encode/1,
+         process/2,
+         process_stream/3,
+         checksum_binary_to_term/1,
+         term_to_checksum_binary/1
+        ]).
+
+-record(state, {client}).
+
+%% @doc init/0 callback. Returns the service internal start
+%% state.
+-spec init() -> any().
+init() ->
+    {ok, C} = riak:local_client(),
+    #state{client=C}.
+
+%% @doc decode/2 callback. Decodes an incoming message.
+decode(Code, Bin) ->
+    Msg = riak_pb_codec:decode(Code, Bin),
+    case Msg of
+        #rpbcoveragereq{type=T, bucket=B} ->
+            Bucket = bucket_type(T, B),
+            {ok, Msg, {"riak_kv.cover", Bucket}}
+    end.
+
+%% @doc encode/1 callback. Encodes an outgoing response message.
+encode(Message) ->
+    {ok, riak_pb_codec:encode(Message)}.
+
+%% @doc process_stream/3 callback. This service does not create any
+%% streaming responses and so ignores all incoming messages.
+process_stream(_,_,State) ->
+    {ignore, State}.
+
+%% @doc process/2 callback. Handles an incoming request message.
+process(#rpbcoveragereq{type=T, bucket=B, min_partitions=P, replace_cover=undefined}, #state{client=Client} = State) ->
+    Bucket = bucket_type(T, B),
+    convert_list(Client:get_cover(Bucket, P), State);
+process(#rpbcoveragereq{type=T, bucket=B, min_partitions=P, replace_cover=R, unavailable_cover=U}, #state{client=Client} = State) ->
+    Bucket = bucket_type(T, B),
+    convert_list(
+      Client:replace_cover(
+        Bucket, P,
+        checksum_binary_to_term(R),
+        lists:map(fun checksum_binary_to_term/1, U)),
+      State).
+
+-spec term_to_checksum_binary(term()) -> term().
+term_to_checksum_binary(C) ->
+    Csum = erlang:adler32(term_to_binary(C)),
+    term_to_binary({Csum, C}).
+
+-spec checksum_binary_to_term(term()) -> {ok, term()} |
+                                         {'error', atom()}.
+checksum_binary_to_term(C) ->
+    verify_checksum_tuple(binary_to_term(C)).
+
+verify_checksum_tuple({Csum, Term}) ->
+    case Csum =:= erlang:adler32(term_to_binary(Term)) of
+        true ->
+            {ok, Term};
+        false ->
+            {error, invalid_checksum}
+    end;
+verify_checksum_tuple(_) ->
+    {error, invalid_structure}.
+
+convert_list({error, Error}, State) ->
+    {error, Error, State};
+convert_list(Results, State) ->
+    %% Pull hostnames & ports
+    %% Wrap each element of this list into a rpbcoverageentry
+    Resp = #rpbcoverageresp{
+              entries=
+                  lists:map(fun(P) ->
+                      Node = proplists:get_value(node, P),
+                      Desc = format_coverage_md(
```

Comment:
Do you think a reminder here that the `Desc` is intended to be human readable would be helpful?